Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -556,35 +557,97 @@ public class UnsubscribePacket {
                  userProperties: [UserProperty]? = nil) {
         self.topicFilters = topicFilters
         self.userProperties = userProperties
+
+        nativeTopicFilters = convertTopicFilters(self.topicFilters)
     }
 
     // Allow an UnsubscribePacket to be created directly using a single topic filter
     public convenience init (topicFilter: String,
                              userProperties: [UserProperty]? = nil) {
-            self.init(topicFilters: [topicFilter],
-                      userProperties: userProperties)
+        self.init(topicFilters: [topicFilter], userProperties: userProperties)
+    }
+
+    typealias RawType = aws_mqtt5_packet_unsubscribe_view
+    func withCStruct<Result>(_ body: (RawType) -> Result) -> Result {
+        var raw_unsubscribe_view = aws_mqtt5_packet_unsubscribe_view()
+        raw_unsubscribe_view.topic_filters = UnsafePointer(nativeTopicFilters)
+        raw_unsubscribe_view.topic_filter_count = topicFilters.count
+        return withOptionalUserPropertyArray(of: userProperties) { userPropertyPointer in
+                if let _userPropertyPointer = userPropertyPointer {
+                    raw_unsubscribe_view.user_property_count = userProperties!.count
+                    raw_unsubscribe_view.user_properties =
+                        UnsafePointer<aws_mqtt5_user_property>(_userPropertyPointer)
+            }
+            return body(raw_unsubscribe_view)
+        }
+    }
+
+    func convertTopicFilters(_ topicFilters: [String]) -> UnsafeMutablePointer<aws_byte_cursor>? {
+        let cArray = UnsafeMutablePointer<aws_byte_cursor>.allocate(capacity: topicFilters.count)
+
+        for (index, string) in topicFilters.enumerated() {
+            let data = string.data(using: .utf8)!
```

Comment:
Concerns about naked unwraps.  We never want to crash on checkable validation errors.

Reply:
We are planning to address these as well as any other Swift -> Native validation checks per operation in a separate branch.